# 1.move
```bash
SUBSAMPLE_IDS=("06" "69" "70" "71" "72" "73" "74" "75" "76" "77" "78" "79" "80" "81" "82" "83" "84" "85" "86" "87" "88" "89" "90" "91")
N_SUBSAMPLE_IDS="${#SUBSAMPLE_IDS[@]}"
for ((IDX=0; IDX<${N_SUBSAMPLE_IDS}; IDX+=1)); do
SUBSAMPLE_ID="${SUBSAMPLE_IDS[IDX]}"
rsync -av --progress desmarais_ill@bmc-150.mit.edu:/mnt/bmc-pub17/DesMarais/211112Des/D21-3180${SUBSAMPLE_ID}-5474R/211112Des_D21-3180${SUBSAMPLE_ID}_*_sequence.fastq.gz /Volumes/TOSHIBA\ EXT/all_raw/raw_ASE
done
```

# 2.Unzip 
```bash
gunzip *_sequence.fastq.gz 
```

# 3.Trim
```bash
#!/bin/bash
#!/usr/bin/env bash
SUBSAMPLE_IDS=("06" "69" "70" "71" "72" "73" "74" "75" "76" "77" "78" "79" "80" "81" "82" "83" "84" "85" "86" "87" "88" "89" "90" "91")
#N_SUBSAMPLE_IDS="${#SUBSAMPLE_IDS[@]}"
#for ((IDX=0; IDX<${N_SUBSAMPLE_IDS}; IDX+=1)); do
SUBSAMPLE_ID=${SUBSAMPLE_IDS[$1]}"
conda activate bowtie
	SEQ_FILE_R1="/home/gridsan/jyun/rawreads/211112Des_D21-3180${SUBSAMPLE_ID}_1_sequence.fastq"
	SEQ_FILE_R2="/home/gridsan/jyun/rawreads/211112Des_D21-3180${SUBSAMPLE_ID}_2_sequence.fastq"
	cd /home/gridsan/jyun/rawreads/	
	trimmomatic PE "${SEQ_FILE_R1}" "${SEQ_FILE_R2}" "${SUBSAMPLE_ID}_-1_trimmed1.fastq"  "${SUBSAMPLE_ID}_-1_trimmed1_unpaired.fastq" "${SUBSAMPLE_ID}_-2_trimmed1.fastq" "${SUBSAMPLE_ID}_-2_trimmed1_unpaired.fastq" ILLUMINACLIP:/home/gridsan/jyun/.conda/envs/bowtie/share/trimmomatic-0.38-1/adapters/TruSeq3-PE-2.fa:2:30:10:1:true
	export PATH=~/.local/bin/:$PATH
	cutadapt --quiet -a A{20} -m 75 -o "${SUBSAMPLE_ID}_-1_trimmed2.fastq" -p "${SUBSAMPLE_ID}_-2_trimmed2.fastq" "${SUBSAMPLE_ID}_-1_trimmed1.fastq" "${SUBSAMPLE_ID}_-2_trimmed1.fastq"
```

# 4.STAR mapping 
## generate genome
```bash
STAR --runThreadN 6 \
--runMode genomeGenerate \
--genomeDir star_genome \
--genomeFastaFiles /home/gridsan/jyun/ase_mase/Bdistachyon_556_v3.0.fa  \
--sjdbGTFfile /home/gridsan/jyun/ase_mase/Bdistachyon_556_v3.2.gene_complete.gtf  \
--sjdbOverhang 99
```
## two pass with strict parameters
```bash 
#!/bin/bash
#!/usr/bin/env bash
SUBSAMPLE_IDS=("06" "69" "70" "71" "72" "73" "74" "75" "76" "77" "78" "79" "80" "81" "82" "83" "84" "85" "86" "87" "88" "89" "90" "91")
#N_SUBSAMPLE_IDS="${#SUBSAMPLE_IDS[@]}"
#for ((IDX=0; IDX<${N_SUBSAMPLE_IDS}; IDX+=1)); do
SUBSAMPLE_ID="${SUBSAMPLE_IDS[$1]}"
STAR --genomeDir /home/gridsan/jyun/ase_mase/star_genome/ \
--runThreadN 6 \
--readFilesIn /home/gridsan/jyun/rawreads/${SUBSAMPLE_ID}_-1_trimmed2.fastq /home/gridsan/jyun/rawreads/${SUBSAMPLE_ID}_-2_trimmed2.fastq  \
--outFileNamePrefix /home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_1p_ \
--outSAMtype BAM SortedByCoordinate \
--outSAMunmapped Within \
--outSAMattributes Standard \
--alignSJoverhangMin 30 \
--alignSJDBoverhangMin 10
STAR --genomeDir /home/gridsan/jyun/ase_mase/star_genome/ \
--runThreadN 6 \
--readFilesIn /home/gridsan/jyun/rawreads/${SUBSAMPLE_ID}_-1_trimmed2.fastq /home/gridsan/jyun/rawreads/${SUBSAMPLE_ID}_-2_trimmed2.fastq  \
--outFileNamePrefix /home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_2p_ \
--outSAMtype BAM SortedByCoordinate \
--outSAMunmapped Within \
--outSAMattributes Standard \
--alignSJoverhangMin 30 \
--alignSJDBoverhangMin 10 \
--sjdbFileChrStartEnd /home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_1p_SJ.out.tab 
```
## remove duplicate
```bash
#!/bin/bash
#!/usr/bin/env bash
SUBSAMPLE_IDS=("06" "69" "70" "71" "72" "73" "74" "75" "76" "77" "78" "79" "80" "81" "82" "83" "84" "85" "86" "87" "88" "89" "90" "91")
#N_SUBSAMPLE_IDS="${#SUBSAMPLE_IDS[@]}"
#for ((IDX=0; IDX<${N_SUBSAMPLE_IDS}; IDX+=1)); do
SUBSAMPLE_ID="${SUBSAMPLE_IDS[$1]}"
cd
conda deactivate 
java -jar picard.jar MarkDuplicates \
  I\=/home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_2p_Aligned.sortedByCoord.out.bam  \
  O\=/home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_2p_Aligned.sortedByCoord.out_dup.bam  \
  M\=/home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_2p_Aligned.sortedByCoord.out.bam.txt \
  VALIDATION_STRINGENCY\=SILENT \
  REMOVE_DUPLICATES\=true \
TMP_DIR\=/home/gridsan/jyun/gatk_bams
```
## gatk find indel and SNPs
```bash
#!/bin/bash
#!/usr/bin/env bash
SUBSAMPLE_IDS=("06" "69" "70" "71" "72" "73" "74" "75" "76" "77" "78" "79" "80" "81" "82" "83" "84" "85" "86" "87" "88" "89" "90" "91")
#N_SUBSAMPLE_IDS="${#SUBSAMPLE_IDS[@]}"
#for ((IDX=0; IDX<${N_SUBSAMPLE_IDS}; IDX+=1)); do
SUBSAMPLE_ID="${SUBSAMPLE_IDS[$1]}"
cd
cd gatk-4.2.5.0
./gatk --java-options -Xmx256G HaplotypeCaller --emit-ref-confidence GVCF -R /home/gridsan/jyun/ase_mase/Bdistachyon_556_v3.0.fa  -I /home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam -O /home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.g.vcf --dont-use-soft-clipped-bases
```
## gatk find indel and SNPs and filter
```bash
#!/bin/bash
#!/usr/bin/env bash
SUBSAMPLE_IDS=("06" "69" "70" "71" "72" "73" "74" "75" "76" "77" "78" "79" "80" "81" "82" "83" "84" "85" "86" "87" "88" "89" "90" "91")
#N_SUBSAMPLE_IDS="${#SUBSAMPLE_IDS[@]}"
#for ((IDX=0; IDX<${N_SUBSAMPLE_IDS}; IDX+=1)); do
SUBSAMPLE_ID="${SUBSAMPLE_IDS[$1]}"
conda deactivate
cd gatk-4.2.5.0
./gatk --java-options -Xmx256G HaplotypeCaller -R /home/gridsan/jyun/ase_mase/Bdistachyon_556_v3.0.fa  -I /home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam -O /home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.vcf --dont-use-soft-clipped-bases
./gatk VariantFiltration \
   -V /home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.vcf \
   -O /home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter.vcf \
   --filter-name "QD2" \
   --filter-expression "QD < 2.0" \
   --filter-name "QUAL30" \
   --filter-expression "QUAL < 30.0"\
   --filter-name "SOR3" \
   --filter-expression "SOR > 3.0" \
   --filter-name "FS60" \
   --filter-expression "FS > 60.0" \
   --filter-name "MQ20" \
   --filter-expression "MQ < 20.0" \
   --filter-name "MQRankSum-12.5" \
   --filter-expression "MQRankSum < -12.5"\
   --filter-name "ReadPosRankSum-8.0" \
   --filter-expression "ReadPosRankSum < -8.0"
module load anaconda/2020a
conda activate bowtie
cd
conda activate bowtie
cat /home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter.vcf  | vcffilter -f "FILTER = PASS"|vcffilter -g "GT = 1/1" | vcffixup - | vcffilter -f "AC > 0"  > /home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter2.vcf
```
## filter again with vcffilter and index
```bash
#!/bin/bash
#!/usr/bin/env bash
SUBSAMPLE_IDS=("06" "69" "70" "71" "72" "73" "74" "75" "76" "77" "78" "79" "80" "81" "82" "83" "84" "85" "86" "87" "88" "89" "90" "91")
#N_SUBSAMPLE_IDS="${#SUBSAMPLE_IDS[@]}"
#for ((IDX=0; IDX<${N_SUBSAMPLE_IDS}; IDX+=1)); do
SUBSAMPLE_ID="${SUBSAMPLE_IDS[$1]}"
cd
conda activate bowtie
cat /home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter.vcf  | vcffilter -f "FILTER = PASS"|vcffilter -g "GT = 1/1" | vcffixup - | vcffilter -f "AC > 0"  > /home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter2.vcf
bgzip -c /home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter2.vcf > /home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter2.vcf.gz
bcftools index -f /home/gridsan/jyun/rawreads/star_alignment/${SUBSAMPLE_ID}_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter2.vcf.gz
```
## save common snps with more than half of the samples
```
cd /home/gridsan/jyun/rawreads/star_alignment/
conda activate bowtie
bcftools isec -p isec_Bd31_output_2+ -n +2 -c all /home/gridsan/jyun/rawreads/star_alignment/80_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter2.vcf.gz  /home/gridsan/jyun/rawreads/star_alignment/78_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter2.vcf.gz /home/gridsan/jyun/rawreads/star_alignment/87_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter2.vcf.gz /home/gridsan/jyun/rawreads/star_alignment/89_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter2.vcf.gz
bcftools isec -p isec_Bd21_output_3+ -n +3 -c all /home/gridsan/jyun/rawreads/star_alignment/06_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter2.vcf.gz  /home/gridsan/jyun/rawreads/star_alignment/69_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter2.vcf.gz /home/gridsan/jyun/rawreads/star_alignment/73_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter2.vcf.gz /home/gridsan/jyun/rawreads/star_alignment/75_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter2.vcf.gz /home/gridsan/jyun/rawreads/star_alignment/82_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter2.vcf.gz /home/gridsan/jyun/rawreads/star_alignment/84_str_2p_Aligned.sortedByCoord.out_dup_split_grp_srt.bam.filter2.vcf.gz
###/home/gridsan/jyun/gatk_bams/bd21tobd21_genome_300_filtered2_complete.vcf is just to hold some chromosome with no snaps
cd isec_Bd31_output_2+
bgzip -c 0000.vcf > 0000.vcf.gz
bcftools index -f 0000.vcf.gz
bgzip -c 0001.vcf > 0001.vcf.gz
bcftools index -f 0001.vcf.gz
bgzip -c 0002.vcf > 0002.vcf.gz
bcftools index -f 0002.vcf.gz
bgzip -c 0003.vcf > 0003.vcf.gz
bcftools index -f 0003.vcf.gz
bgzip -c  /home/gridsan/jyun/gatk_bams/bd21tobd21_genome_300_filtered2_complete.vcf > /home/gridsan/jyun/gatk_bams/bd21tobd21_genome_300_filtered2_complete.vcf.gz
####
bcftools index -f /home/gridsan/jyun/gatk_bams/bd21tobd21_genome_300_filtered2_complete.vcf.gz
bcftools merge 0000.vcf.gz 0001.vcf.gz 0002.vcf.gz 0003.vcf.gz /home/gridsan/jyun/gatk_bams/bd21tobd21_genome_300_filtered2_complete.vcf.gz --force-samples > Bd31_all.vcf
cd /home/gridsan/jyun/rawreads/star_alignment/isec_Bd21_output_3+
bgzip -c 0000.vcf > 0000.vcf.gz
bcftools index -f 0000.vcf.gz
bgzip -c 0001.vcf > 0001.vcf.gz
bcftools index -f 0001.vcf.gz
bgzip -c 0002.vcf > 0002.vcf.gz
bcftools index -f 0002.vcf.gz
bgzip -c 0003.vcf > 0003.vcf.gz
bcftools index -f 0003.vcf.gz
bgzip -c 0004.vcf > 0004.vcf.gz
bcftools index -f 0004.vcf.gz
bgzip -c 0005.vcf > 0005.vcf.gz
bcftools index -f 0005.vcf.gz
bcftools merge 0000.vcf.gz 0001.vcf.gz 0002.vcf.gz 0003.vcf.gz 0004.vcf.gz 0005.vcf.gz /home/gridsan/jyun/gatk_bams/bd21tobd21_genome_300_filtered2_complete.vcf.gz --force-samples > Bd21_all.vcf
Replace "./.:.:.:.:." with "1/1:0,10:10:30:281,30,0" in Bd21_all.vcf using nano
And Bd31
conda deactivate 
cd
module load anaconda/2022a
cd ase_mase/test_loose_setting
```
## liftover 
```bash
source activate g2gtools
g2gtools vcf2vci -o /home/gridsan/jyun/ase_mase/test_loose_setting/BD31_genome.vci -s bd31 -i /home/gridsan/jyun/rawreads/star_alignment/isec_Bd31_output_2+/Bd31_all.vcf -f /home/gridsan/jyun/ase_mase/Bdistachyon_556_v3.0.fa 
g2gtools vcf2vci -o /home/gridsan/jyun/ase_mase/test_loose_setting/BD21_genome.vci -s bd21 -i /home/gridsan/jyun/rawreads/star_alignment/isec_Bd21_output_3+/Bd21_all.vcf -f /home/gridsan/jyun/ase_mase/Bdistachyon_556_v3.0.fa
g2gtools vcf2vci -o /home/gridsan/jyun/ase_mase/test_loose_setting/BD31_genome.vci -s bd31 -i /home/gridsan/jyun/rawreads/star_alignment/isec_Bd31_output_2+/Bd31_all.vcf.gz -f /home/gridsan/jyun/ase_mase/Bdistachyon_556_v3.0.fa 
g2gtools vcf2vci -o /home/gridsan/jyun/ase_mase/test_loose_setting/BD21_genome.vci -s bd21 -i /home/gridsan/jyun/rawreads/star_alignment/isec_Bd21_output_3+/Bd21_all.vcf.gz -f /home/gridsan/jyun/ase_mase/Bdistachyon_556_v3.0.fa
gzip -dk /home/gridsan/jyun/rawreads/star_alignment/isec_Bd31_output_2+/Bd31_all.vcf.gz
gzip -dk /home/gridsan/jyun/rawreads/star_alignment/isec_Bd21_output_3+/Bd21_all.vcf.gz
export PATH=/home/gridsan/jyun/ase_mase/bin:$PATH
tabix -p vcf -f BD31_genome.vci.gz
tabix -p vcf -f BD21_genome.vci.gz
g2gtools patch -i /home/gridsan/jyun/ase_mase/Bdistachyon_556_v3.0.fa -c /home/gridsan/jyun/ase_mase/test_loose_setting/BD21_genome.vci.gz -o  /home/gridsan/jyun/ase_mase/test_loose_setting/BD21_genome.PATCHED.fa 
g2gtools transform -i /home/gridsan/jyun/ase_mase/test_loose_setting/BD21_genome.PATCHED.fa -c /home/gridsan/jyun/ase_mase/test_loose_setting/BD21_genome.vci.gz -o /home/gridsan/jyun/ase_mase/test_loose_setting/BD21_genome.fa
g2gtools patch -i /home/gridsan/jyun/ase_mase/Bdistachyon_556_v3.0.fa -c /home/gridsan/jyun/ase_mase/test_loose_setting/BD31_genome.vci.gz -o  /home/gridsan/jyun/ase_mase/test_loose_setting/BD31_genome.PATCHED.fa  
g2gtools transform -i /home/gridsan/jyun/ase_mase/test_loose_setting/BD31_genome.PATCHED.fa -c /home/gridsan/jyun/ase_mase/test_loose_setting/BD31_genome.vci.gz -o /home/gridsan/jyun/ase_mase/test_loose_setting/BD31_genome.fa
g2gtools convert -i /home/gridsan/jyun/ase_mase/Bdistachyon_556_v3.2.gene_complete.gtf -c /home/gridsan/jyun/ase_mase/test_loose_setting/BD31_genome.vci.gz -o /home/gridsan/jyun/ase_mase/test_loose_setting/BD31_genome.gtf 
g2gtools convert -i /home/gridsan/jyun/ase_mase/Bdistachyon_556_v3.2.gene_complete.gtf -c /home/gridsan/jyun/ase_mase/test_loose_setting/BD21_genome.vci.gz -o /home/gridsan/jyun/ase_mase/test_loose_setting/BD21_genome.gtf 
##check unmapped.gtf here
source deactivate
exit
```



## send back
```bash
rsync -av jyun@txe1-login.mit.edu:/home/gridsan/jyun/ase_mase/test_loose_setting/BD31_genome.fa /Volumes/TOSHIBA_backup3/ASE_ALN2/final_use/test_loose_setting
rsync -av jyun@txe1-login.mit.edu:/home/gridsan/jyun/ase_mase/test_loose_setting/BD21_genome.fa /Volumes/TOSHIBA_backup3/ASE_ALN2/final_use/test_loose_setting
rsync -av jyun@txe1-login.mit.edu:/home/gridsan/jyun/ase_mase/test_loose_setting/BD31_genome.gtf /Volumes/TOSHIBA_backup3/ASE_ALN2/final_use/test_loose_setting
rsync -av jyun@txe1-login.mit.edu:/home/gridsan/jyun/ase_mase/test_loose_setting/BD21_genome.gtf /Volumes/TOSHIBA_backup3/ASE_ALN2/final_use/test_loose_setting

python ~/Downloads/emase3/scripts/prepare-emase.py -G /Volumes/TOSHIBA_backup3/ASE_ALN2/final_use/Bdistachyon_556_v3.0.fa -g /Volumes/TOSHIBA_backup3/ASE_ALN2/final_use/Bdistachyon_556_v3.2.gene_complete.gtf -o /Volumes/TOSHIBA_backup3/ASE_ALN2/final_use/test_loose_setting/ -m --no-bowtie-index 2> file2.log
python ~/Downloads/emase3/scripts/prepare-emase.py -G /Volumes/TOSHIBA_backup3/ASE_ALN2/final_use/test_loose_setting/BD21_genome.fa,/Volumes/TOSHIBA_backup3/ASE_ALN2/final_use/test_loose_setting/BD31_genome.fa -s bd21,bd31 -o /Volumes/TOSHIBA_backup3/ASE_ALN2/final_use/test_loose_setting/ 2> file2.log

rsync -av  /Volumes/TOSHIBA_backup3/ASE_ALN2/final_use/test_loose_setting/emase.gene2transcripts_combinebd2131.tsv jyun@txe1-login.mit.edu:/home/gridsan/jyun/ase_mase/test_loose_setting/


rsync -av /Volumes/TOSHIBA_backup3/ASE_ALN2/final_use/test_loose_setting/emase.* jyun@txe1-login.mit.edu:/home/gridsan/jyun/ase_mase/test_loose_setting/
rsync -av /Volumes/TOSHIBA_backup3/ASE_ALN2/final_use/test_loose_setting/bowtie.* jyun@txe1-login.mit.edu:/home/gridsan/jyun/ase_mase/test_loose_setting/
rsync -av jyun@txe1-login.mit.edu:/home/gridsan/jyun/ase_mase/trim_fastq/test_1_sort.bam* /Volumes/TOSHIBA_backup3/ASE_ALN2/final_use/
```

## run alignment on the deploid genome
```bash
#!/bin/bash
#!/usr/bin/env bash
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=1
#SBATCH -a 0-23
./run.sh $SLURM_ARRAY_TASK_ID
```
```bash
#!/bin/bash
#!/usr/bin/env bash
SUBSAMPLE_IDS=("06" "69" "70" "71" "72" "73" "74" "75" "76" "77" "78" "79" "80" "81" "82" "83" "84" "85" "86" "87" "88" "89" "90" "91")
#N_SUBSAMPLE_IDS="${#SUBSAMPLE_IDS[@]}"
#for ((IDX=0; IDX<${N_SUBSAMPLE_IDS}; IDX+=1)); do
SUBSAMPLE_ID="${SUBSAMPLE_IDS[$1]}"
module load anaconda/2022a 
conda activate bowtie
bowtie -q -a --best --strata --sam -v 3 -m 100 /home/gridsan/jyun/ase_mase/test_loose_setting/bowtie.transcripts /home/gridsan/jyun/rawreads/${SUBSAMPLE_ID}_-1_trimmed2.fastq -S /home/gridsan/jyun/ase_mase/trim_fastq/${SUBSAMPLE_ID}_1.sam
bowtie  -q -a --best --strata --sam -v 3 -m 100 /home/gridsan/jyun/ase_mase/test_loose_setting/bowtie.transcripts /home/gridsan/jyun/rawreads/${SUBSAMPLE_ID}_-2_trimmed2.fastq -S /home/gridsan/jyun/ase_mase/trim_fastq/${SUBSAMPLE_ID}_2.sam
cd /home/gridsan/jyun/ase_mase/trim_fastq
python3 process_${SUBSAMPLE_ID}.py
#samtools
export PATH=/home/gridsan/jyun/samtoolsinstall/bin:$PATH
samtools view -h -o ${SUBSAMPLE_ID}_combined.bam ${SUBSAMPLE_ID}_combined.sam
samtools rmdup -s ${SUBSAMPLE_ID}_combined.bam ${SUBSAMPLE_ID}_combined_rmdup.bam
#done
exit 0
```
## prepare files to run emase
```bash
conda activate alntools
SUBSAMPLE_IDS=(  "78" "79" "80" "81" "82" "83" "84" "85" "86" "87" "88" "89" "90" "91")
N_SUBSAMPLE_IDS="${#SUBSAMPLE_IDS[@]}"
for ((IDX=0; IDX<${N_SUBSAMPLE_IDS}; IDX+=1)); do
SUBSAMPLE_ID="${SUBSAMPLE_IDS[IDX]}"
rsync jyun@txe1-login.mit.edu:/home/gridsan/jyun/ase_mase/trim_fastq/${SUBSAMPLE_ID}_combined_rmdup.bam /Volumes/TOSHIBA_backup3/ASE_ALN2/final_rna_emase
alntools bam2ec -c 1 /Volumes/TOSHIBA_backup3/ASE_ALN2/final_rna_emase/${SUBSAMPLE_ID}_combined_rmdup.bam /Volumes/TOSHIBA_backup3/ASE_ALN2/final_rna_emase/${SUBSAMPLE_ID}_combined.bin
done
```
## run emase
```bash
SUBSAMPLE_IDS=("06" "69" "70" "71" "72" "73" "74" "75" "76")
N_SUBSAMPLE_IDS="${#SUBSAMPLE_IDS[@]}"
for ((IDX=0; IDX<${N_SUBSAMPLE_IDS}; IDX+=1)); do
SUBSAMPLE_ID="${SUBSAMPLE_IDS[IDX]}"
cd /Users/jiey/Desktop/untitled\ folder\ 4/churchill-lab-emase-zero-6a439d7/src
./emase-zero -m 4 /Volumes/TOSHIBA_backup3/ASE_ALN2/final_rna_emase/${SUBSAMPLE_ID}_combined.bin -o /Volumes/TOSHIBA_backup3/ASE_ALN2/final_rna_emase/${SUBSAMPLE_ID}_combined.bin.quantified -l /Volumes/TOSHIBA_backup3/ASE_ALN2/final_use/emase.pooled.transcripts.info  -g /Volumes/TOSHIBA_backup3/ASE_ALN2/final_use/emase.gene2transcripts.tsv
```
## send together
```bash
SUBSAMPLE_IDS=("06" "69" "70" "71" "72" "73" "74" "75" "76" "77" "78" "79" "80" "81" "82" "83" "84" "85" "86" "87" "88" "89" "90" "91")
N_SUBSAMPLE_IDS="${#SUBSAMPLE_IDS[@]}"
for ((IDX=0; IDX<=${N_SUBSAMPLE_IDS}; IDX+=1)); do
SUBSAMPLE_ID="${SUBSAMPLE_IDS[IDX]}"
rsync -av /Volumes/TOSHIBAEXT/all_raw/D21-3180${SUBSAMPLE_ID}-5474R/211112Des_D21-3180${SUBSAMPLE_ID}_*_sequence.fastq.gz /Volumes/TOSHIBAEXT/all_raw/RNASeq_ASE/
done
```
## generate md5
```bash
SUBSAMPLE_IDS=("06" "69" "70" "71" "72" "73" "74" "75" "76" "77" "78" "79" "80" "81" "82" "83" "84" "85" "86" "87" "88" "89" "90" "91")
N_SUBSAMPLE_IDS="${#SUBSAMPLE_IDS[@]}"
for ((IDX=0; IDX<=${N_SUBSAMPLE_IDS}; IDX+=1)); do
SUBSAMPLE_ID="${SUBSAMPLE_IDS[IDX]}"
md5 /Volumes/TOSHIBAEXT/all_raw/RNASeq_ASE/211112Des_D21-3180${SUBSAMPLE_ID}_1_sequence.fastq.gz 
md5 /Volumes/TOSHIBAEXT/all_raw/RNASeq_ASE/211112Des_D21-3180${SUBSAMPLE_ID}_2_sequence.fastq.gz 
done
```

